### PR TITLE
SHIP-0002 - Deprecate the Runtime Image Feature

### PR DESCRIPTION
**Pull request by imjasonh**
@adambkaplan 
@qu1queee 

cc @otaviof @sbose78

### CODE REVIEW

**Comment by imjasonh** [comment]
To get a sense of what code changes might look like:

- WIP diff to add deprecation warnings: https://github.com/shipwright-io/build/compare/main...imjasonh:deprecate-runtime
- WIP diff to remove the feature: https://github.com/shipwright-io/build/compare/main...imjasonh:remove-runtime

File: ships/deprecate-runtime.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+### Goals
+
+Deter new users from adopting `runtime`, until we are comfortable removing the feature entirely.
+
+### Non-Goals
+
+Replace every aspect of the feature with new Shipwright-level features.
+
+- Some features should be the responsibility of the build tool (e.g., multi-stage docker builds, Buildpacks), and Shipwright should encourage and document them.
+- Some features might end up becoming other more focused Shipwright-level features (e.g., rebasing an image)a.
```
Comment (qu1queee) [inline ships/deprecate-runtime.md:78]:
```suggestion
- Some features might end up becoming other more focused Shipwright-level features (e.g., rebasing an image)
```

**Review by qu1queee — CHANGES_REQUESTED** [review]
This looks good to me. I dont think there is more to say than what you are expressing in here. I would like to approve this asap. There is just a very small typo I think.

**Review by qu1queee — APPROVED** [review]
/approve

File: ships/deprecate-runtime.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+- Deprecate the feature
+
+### Risks and Mitigations
+
+At this time we don't believe the feature has significant usage, but due to the nature of installations, there's no way to be sure.
+While the feature is marked as deprecated, we might find a user significantly depends on it, and we should be open to considering un-deprecating it, or modifying our removal timeline/process as necessary.
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
```
Comment (adambkaplan) [inline ships/deprecate-runtime.md:116]: Drawback - we are removing a feature without a direct replacement.

File: ships/deprecate-runtime.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+What features of `runtime` do we expect future users to want, and how should we deliver them in a manner that's consistent with the future direction of the API?
```
Comment (adambkaplan) [inline ships/deprecate-runtime.md:36]: Should we mark this SHIP as "provisional" as long as this question is open?

**Review by adambkaplan — COMMENTED** [review]
@imjasonh Please update the title of this to "SHIP-0002 - Deprecate the Runtime Image Feature", and likewise rename the filename to `ship-0002-deprecate-runtime.md`

File: ships/deprecate-runtime.md (hunk 3 above)
Comment (imjasonh) [reply ships/deprecate-runtime.md]: Good question. I would bias toward action here, even if there are unanswered questions. While the feature is live and non-deprecated, we risk people discovering it and coming to depend on it, which will make deprecating and removing harder later.

We can delay removing it for as long as we feel necessary, but we should deprecate ASAP if we think we ever will (and I believe we have consensus on that much at least).

**Comment by imjasonh** [comment]
> @imjasonh Please update the title of this to "SHIP-0002 - Deprecate the Runtime Image Feature", and likewise rename the filename to `ship-0002-deprecate-runtime.md`

Done. This is good advice for future SHIPs, I'll send a PR to add it to guidelines.md

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**